src/ui/webeiddialog.cpp

```diff
@@ -287,7 +290,11 @@ QString WebEidDialog::getPin()
 {
     // getPin() is called from background threads and must be thread-safe.
     // QString uses QAtomicPointer internally and is thread-safe.
-    return pin;
+    // There should be only single reference and this is transferred to the caller for safety
+    QString ret = pin;
+    eraseData(pin);
+    pin.clear();
+    return ret;
```

Comment:
```suggestion
    // There should be only single reference and this is transferred to the caller for safety
    return std::move(pin);
```

Reply:
I think this is better alternative

Reply:
AFAIK std::move gives only hint, not guarantee, of non-duplication of the data.
All sane implementations do the right thing, but IMO we should rely on specification, not implementation.
I may be wrong, but then I'd say that the specification is confusing, and that is not a good thing for security critical code